public/js/helper.js

```diff
@@ -1,45 +1,22 @@
-function clone(obj) {
-	return JSON.parse(JSON.stringify(obj));
-}
+const clone = obj => JSON.parse(JSON.stringify(obj));
 
-function isEmpty(obj) {
-    for(var key in obj) {
-        if(obj.hasOwnProperty(key))
-            return false;
-    }
-    return true;
-}
+// todo: I think Object.keys(obj).some(_=>_) should work as well.
+const isEmpty = obj => Object.keys(obj).some(key => obj.hasOwnProperty(key));
 
-function arrayRemove(arr, value) {
-	return arr.filter(function(ele) {
-	   return ele != value;
-	});
-}
+const arrayRemove = (arr, value) => arr.filter(ele => ele != value);
 
-function guid() {
-	function s4() {
-		return Math.floor((1 + Math.random()) * 0x10000)
-		  .toString(16)
-		  .substring(1);
-	}
-	return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() + s4() + s4();
-}
+const s4 = () => Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
```

Comment:
Maybe choose another name if it is declared outside of guid and shortgui

Reply:
Yes it would be better. Any proposition?

Reply:
Idk, randStr4? :p